remote-view/shared/src/main/scala/parsley/debug/RemoteView.scala

```diff
@@ -0,0 +1,128 @@
+    println("Sending Debug Tree to Server")
+    
+    val backend = TryHttpURLConnectionBackend(
+      options = SttpBackendOptions.connectionTimeout(ConnectionTimeout)
+    )
+
+    val response: Try[Response[Either[String,String]]] = basicRequest
+      .readTimeout(ResponseTimeout)
+      .header("User-Agent", "remoteView")
+      .contentType("application/json")
+      .body(payload)
+      .post(endPoint)
+      .send(backend)
+
+    response match {
+      case Failure(exception) => println(s"${TextToRed}Remote View request failed! Please validate address ($address) and port number ($port).${TextToNormal}\n\tError : ${exception.toString()}")
+      case Success(value) => value.body match {
+        // Left indicates the request is successful, but the response code was not 2xx.
+        case Left(errorMessage) => println(s"${TextToRed}Request Failed with message : $errorMessage, and status code : ${value.code}${TextToNormal}")
+        // Right indicates a successful request with 2xx response code.
+        case Right(body) => println(s"Request successful with message : $body")
+      }
+    }
+  }
+}
+
+object RemoteView extends DebugView.Reusable with RemoteView {
+    // Default port uses HTTP port and local host
+    override protected val port: Int = 80
+    override protected val address: String = "127.0.0.1"
+    
+    private final val MinimalIpLength: Int = "0.0.0.0".length
+    private final val MaximalIpLength: Int = "255.255.255.255".length
+    
+    private final val MaxUserPort: Integer = 0xFFFF
+
+    /** Do some basic validations for a given IP address. */
+    private def checkIp(address: String): Boolean = {
+      val addrLenValid: Boolean = address.length >= MinimalIpLength && address.length <= MaximalIpLength
+      val addrDotValid: Boolean = address.count(_ == '.') == 3
+      
+      // Check that every number is a number
+      val numberStrings: Array[String] = address.split('.')
+      val addrNumValid: Boolean = numberStrings.forall((number: String) => number.length > 0 && {
+          number.toIntOption match {
+            case None => false
+            case Some(number) => number >= 0x0 && number <= 0xFF
+          }
+        }
+      )
+
+      addrLenValid && addrDotValid && addrNumValid 
+    }
+
+    /** Create a new instance of [[RemoteView]] with a given custom port. */
+    def apply(userPort: Integer = port, userAddress: String = address): RemoteView = new RemoteView {
+      require(userPort <= MaxUserPort, s"Remote View port invalid : $userPort > $MaxUserPort")
+      require(checkIp(userAddress), s"Remote View address invalid : $userAddress")
+
+      override implicit val port = userPort
+      override implicit val address = userAddress
+    }
+}
+
+/** Helper object for connecting to the DILL backend. */
+object DillRemoteView extends DebugView.Reusable with RemoteView {
+  // Default endpoint for DILL backend is port 0x444C ("DL") on localhost
+  override protected val port: Int = 0x444C
```

Comment:
```suggestion
  override protected val port: Int = 17484
```
port numbers aren't usually written in hex (I suggested the `0xFFFF` above purely because it's a very well known constant for `2^16`)

Reply:
Yeah, makes sense